plugins/com.google.cloud.tools.eclipse.sdk.ui.test/src/com/google/cloud/tools/eclipse/sdk/ui/preferences/CloudSdkPreferenceAreaTest.java

```diff
@@ -155,6 +156,23 @@ public void testUi_sdkManagementFeature() {
     assertNotNull(sdkLocation);
     assertNotNull(sdkVersion);
   }
+  
+  @Test
+  public void testI2897() {
+    when(cloudSdkManager.isManagedSdkFeatureEnabled()).thenReturn(true);
+    when(preferences.getString(CloudSdkPreferences.CLOUD_SDK_PATH)).thenReturn("");
+    
+    createPreferenceArea();
+    chooseSdk.setSelection(true);
+    sdkLocation.setText("");
+    
+    // now need to invoke update
+    area.updateSelectedVersion();
```

Comment:
```java
    chooseSdk.setSelection(true);
    sdkLocation.setText("");
    area.updateSelectedVersion();
```

All of these three are not needed once you add `    when(preferences.getString(CloudSdkPreferences.CLOUD_SDK_MANAGEMENT)).thenReturn("MANUAL");` as I said earlier.